src/control/cmd/daos_server/storage.go

```diff
@@ -195,10 +217,15 @@ func (cmd *storagePrepareCmd) Execute(args []string) error {
 
 	scanErrors := make(errs, 0, 2)
 
-	if err := cmd.prepNvme(&scanErrors, cmd.scs.NvmePrepare); err != nil {
+	iommuEnabled, err := sysfs.NewProvider(cmd).IsIOMMUEnabled()
```

Comment:
I was thinking you'd pass the sysfs.Provider as a hardware.IOMMUDetector into the prepNvme command, or even set it on the command struct itself. Something like:

if cmd.iommu == nil {
        cmd.iommu = hwprov.DefaultIOMMUDetector(log)
}

With cmd.iommu as a hardware.IOMMUDetector. Then in the tests you could set cmd.iommu with a mock ahead of time, to get either true or false out of it. The more straightforward way in this patch works, too, but making use of the interfaces is the pattern I've been trying to follow with the other hardware providers, with hwprov the only package that has any explicit awareness of the hwloc provider, sysfs, libfabric, etc.

Reply:
sure

Reply:
I've switched it so we only access hwprov rather than sysfs directly, so hopefully that's acceptable. I didn't feel like rewriting the tests just to use the interface would provide any specific benefit so I've left them as this. Given the main objections in the code review have been addressed hopefully I can address any further preferences in a follow on PR?

Reply:
done

Reply:
Okay with me. Thanks Tom.